actionpack/lib/action_controller/metal/live.rb

```diff
@@ -210,6 +202,14 @@ def on_error(&block)
       def call_on_error
         @error_callback.call
       end
+
+      private
+
+        def each_chunk(&block)
+          while str = @buf.pop
```

Comment:
I think we need to retain the `sending!` and `sent!` calls here. Streaming responses are the only place that we shouldn't be able to call `each` multiple times.